foundation/foundation-mda/src/main/java/com/boozallen/aiops/mda/metamodel/element/python/PythonStep.java

```diff
@@ -130,18 +135,25 @@ public Set<String> getImplImports() {
      * @return {@link #imports} that have been appropriately modified to include the correct
      * relative import paths for any references to metamodel-define records or dictionary types.
      */
-    private Set<String> getImportsWithRelativePaths(boolean isImplModule) {
-        Set<String> importsWithRelativePaths = new TreeSet<>();
+    private Set<String> getImports(boolean isImplModule) {
+        Set<String> importsSet = new TreeSet<>();
         for (String moduleImport : imports) {
             if (moduleImport.startsWith("from record.")
                     || moduleImport.startsWith("from dictionary.")) {
-                importsWithRelativePaths.add(moduleImport.replace("from ",
-                        isImplModule ? "from .." : "from ..."));
+                if(profileName.equals("data-delivery-pyspark"))
```

Comment:
A: Given we allow `profileName` to be `null` this is susceptible to a hard to trace NPE. In general, a good practice is to put the constant on the LHS of the equals just in case. E.g. `"data-delivery-pyspark".equals(profileName)`